src/clients/c++/request_common.cc

```diff
@@ -204,6 +205,7 @@ InputImpl::SetRaw(const uint8_t* input, size_t input_byte_size)
 
   bufs_.push_back(input);
   buf_byte_sizes_.push_back(input_byte_size);
+  io_type_ = RAW;
```

Comment:
You need to check at the start of the function if io_type_ is SHARED_MEMORY and if it is return an error.